test/test_bin/test_support_stats_reporting.py

```diff
@@ -0,0 +1,109 @@
+import os
+from datetime import date
+import unittest
+from unittest.mock import patch, MagicMock
+from bin.support_stats_reporting import (
+    SupportRequest,
+    create_dataframe_from_csv,
+    get_yesterdays_support_requests,
+    get_previous_working_day,
+    get_environment_variables,
+    craft_message_to_slack,
+    main,
+)
+
+
+class TestSupportRequestHash(unittest.TestCase):
+
+    def test_support_request_hash(self):
+        support_request1 = SupportRequest("Type A", "Action A", "2023-05-01")
+        support_request2 = SupportRequest("Type B", "Action A", "2023-05-02")
+        self.assertEqual(hash(support_request1), hash(support_request2))
+
+
+class TestCreateDataframeFromCsv(unittest.TestCase):
+
+    @patch("pandas.read_csv")
+    def test_create_dataframe_from_csv(self, mock_read_csv):
+        mock_dataframe = MagicMock()
+        mock_read_csv.return_value = mock_dataframe
+        filepath = "test/fixtures/test_data.csv"
+        result = create_dataframe_from_csv(filepath)
+        mock_read_csv.assert_called_with(filepath)
+        self.assertEqual(result, mock_dataframe)
+
+
+class TestGetPreviousWorkingDay(unittest.TestCase):
+
+    def test_monday_returns_friday(self):
+        previous_day = get_previous_working_day(date_today=date(2024, 7, 8))
+        self.assertEqual(previous_day, str(date(2024, 7, 5)))
+
+    def test_tuesday_returns_monday(self):
+        previous_day = get_previous_working_day(date_today=date(2024, 7, 9))
+        self.assertEqual(previous_day, str(date(2024, 7, 8)))
+
+
+class TestGetYesterdaysSupportRequests(unittest.TestCase):
+
+    @patch("bin.support_stats_reporting.get_previous_working_day")
+    def test_get_yesterdays_support_requests(self, mock_previous_working_day):
+        mock_previous_working_day.return_value = "2023-05-01"
+        all_support_requests = [
+            SupportRequest("Type A", "Action A", "2023-05-01"),
+            SupportRequest("Type B", "Action B", "2023-05-02"),
+            SupportRequest("Type C", "Action C", "2023-05-01"),
+        ]
+        expected_result = [
+            SupportRequest("Type A", "Action A", "2023-05-01"),
+            SupportRequest("Type C", "Action C", "2023-05-01"),
+        ]
+        result = get_yesterdays_support_requests(
+            all_support_requests, todays_date=date(2024, 7, 23)
```

Comment:
```suggestion
            all_support_requests, todays_date=date(2024, 5, 2)
```
To match with the mocked previous date of 2023-05-01?

Reply:
human: 1 - AI: 0 😆

Reply:
Removed the patch and mock, updated the date and test is now working